tests/src/test/scala/org/http4s/parser/UriParserSpec.scala

```diff
@@ -41,7 +33,7 @@ class UriParserSpec extends Http4sSpec {
 
     // http://www.ietf.org/rfc/rfc2396.txt
 
-    "parse a IPv6 address" in {
+    "parse a IPv6 address" in pending {
```

Comment:
not sure why this does not parse.

Is the generator sound?

Reply:
I wondered if maybe the new parser was requiring brackets, but it doesn't look like that's it. :thinking:

Reply:
`value` is skipping leading 0 so it's not the exact same thing.

to do the same thing as the previous test case does we could switch to
```scala
Ipv6Address.parser.string.parseAll(s) must beRight(s)
```

but it cannot parse everything. It is still failing on the following inputs:

```
::32ba:32ba:32ba:32ba:32ba:32ba:32ba  
01ab::32ba:32ba:32ba:32ba:32ba:32ba                                           
01ab:01ab::32ba:32ba:32ba:32ba:32ba
01ab:01ab:01ab::32ba:32ba:32ba:32ba                                           
01ab:01ab:01ab:01ab::32ba:32ba:32ba                                                                                                                          01ab:01ab:01ab:01ab:01ab::32ba:32ba                                           
01ab:01ab:01ab:01ab:01ab:01ab::32ba                                                                                                                          
01ab:01ab:01ab:01ab:01ab:01ab:01ab::
```

so I think the parser might have a small bug

Reply:
```
Index: core/src/main/scala/org/http4s/Uri.scala
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/core/src/main/scala/org/http4s/Uri.scala b/core/src/main/scala/org/http4s/Uri.scala
--- a/core/src/main/scala/org/http4s/Uri.scala	(revision e2d897cda6cfced8759fb0cf87cd7e800a641e14)
+++ b/core/src/main/scala/org/http4s/Uri.scala	(date 1611421411712)
@@ -1195,34 +1195,37 @@
       (repN(6, h16Colon).with1 ~ ls32)
         .map { case (ls: collection.Seq[Short], rs) => toIpv6(ls, rs) }
         .backtrack
-        .orElse((doubleColon *> repN(4, h16Colon, P.unit) ~ ls32)
+        .orElse((doubleColon *> repN(5, h16Colon, P.unit) ~ ls32)
           .map { case (rs: List[Short], rs2) => toIpv6(Seq.empty, rs ++ rs2) })
         .backtrack
-        .orElse(((h16.?.with1 <* doubleColon) ~ h16Colon.repExactlyAs[List[Short]](3).backtrack ~ ls32)
+        .orElse(((h16.?.with1 <* doubleColon) ~ h16Colon.repExactlyAs[List[Short]](4).backtrack ~ ls32)
           .map { case ((ls: Option[Short], rs), rs2) => toIpv6(ls.toSeq, rs ++ rs2) })
         .backtrack
         .orElse(
-          ((repN(2, h16, colon.void).?.with1 <* doubleColon) ~ h16Colon.repExactlyAs[List[Short]](2).backtrack ~ ls32)
+          ((repN(3, h16, colon.void).?.with1 <* doubleColon) ~ h16Colon.repExactlyAs[List[Short]](2).backtrack ~ ls32)
+            .map { case ((ls: Option[List[Short]], rs), rs2) => toIpv6(ls.getOrElse(Seq.empty), rs ++ rs2) })
+        .backtrack
+        .orElse(((repN(2, h16, colon.void).?.with1 <* doubleColon) ~ h16Colon.repExactlyAs[List[Short]](3).backtrack ~ ls32)
             .map { case ((ls: Option[List[Short]], rs), rs2) =>
               toIpv6(ls.getOrElse(Seq.empty), rs ++ rs2)
             })
         .backtrack
-        .orElse(((repN(3, h16, colon.void).?.with1 <* doubleColon) ~ h16Colon ~ ls32)
+        .orElse(((repN(4, h16, colon.void).?.with1 <* doubleColon) ~ h16Colon ~ ls32)
           .map { case ((ls: Option[List[Short]], r0: Short), rs) =>
             toIpv6(ls.getOrElse(Seq.empty), Seq(r0) ++ rs)
           })
         .backtrack
-        .orElse(((repN(4, h16, colon.void).?.with1 <* doubleColon) ~ ls32)
+        .orElse(((repN(5, h16, colon.void).?.with1 <* doubleColon) ~ ls32)
           .map { case (ls: Option[collection.Seq[Short]], rs) =>
             toIpv6(ls.getOrElse(Seq.empty), rs)
           })
         .backtrack
-        .orElse(((repN(5, h16, colon.void).?.with1 <* doubleColon) ~ h16)
+        .orElse(((repN(6, h16, colon.void).?.with1 <* doubleColon) ~ h16)
           .map { case (ls: Option[collection.Seq[Short]], rs: Short) =>
             toIpv6(ls.getOrElse(Seq.empty), Seq(rs))
           })
         .backtrack
-        .orElse((repN(6, h16, colon.void).?.with1 <* doubleColon)
+        .orElse((repN(7, h16, colon.void).?.with1 <* doubleColon)
           .map {(ls: Option[collection.Seq[Short]]) => toIpv6(ls.getOrElse(Seq.empty), Seq.empty)})
         .backtrack
     }
```

makes this test pass but has issues with other tests. but there are a couple off by one errors in the ipv6 parser